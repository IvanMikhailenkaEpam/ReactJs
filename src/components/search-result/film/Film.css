img {
  width: 350px;
  height: 450px
}
.age {
  border: 1px solid black;
  padding: 3px;
}