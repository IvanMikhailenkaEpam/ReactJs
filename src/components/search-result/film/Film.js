import React, { Component } from 'react';
import './Film.css';
import { Link } from 'react-router-dom';
import * as PropTypes from 'prop-types';

class Film extends Component {
  static propTypes = {
    value: PropTypes.object.isRequired
  };

  render() {
    const { value } = this.props;
    return (
      <div className="col-md-4 mt-5">
        <Link to={`/movie/${value.id}`}>
          <img
            src={value.img}
            alt="text"
          />
        </Link>
        <div className="row mt-3">
          <div className="col-6"><span>{value.name}</span></div>
          <div className="col-2 ml-auto mr-3"><span className="age">{value.age}</span></div>
        </div>
        <div><span>{value.genre}</span></div>
      </div>
    );
  }
}

export default Film;
