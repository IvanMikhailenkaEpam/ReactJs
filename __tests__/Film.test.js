import React from 'react';
import {shallow} from 'enzyme';
import Film from "../src/components/search-result/film/Film";

describe('Film', () => {
   const film =
    {
      'id': 1,
      'name': "film 1",
      'genre': 'genre 1',
      'age': 1332,
      'rating': 4.5,
      'img': "https://m.media-amazon.com/images/M/MV5BNGNhMDIzZTUtNTBlZi00MTRlLWFjM2ItYzViMjE3YzI5MjljXkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SY1000_CR0,0,686,1000_AL_.jpg",
      'description': "Some info",
      'director': "Tarantion 1",
      'regards': 'Oscar winning move',
      'duration': '122 min',
    };


  it('render Film correctly', () => {
    const component = shallow(<Film value={film}/>);

    expect(component).toMatchSnapshot();
  });
});
