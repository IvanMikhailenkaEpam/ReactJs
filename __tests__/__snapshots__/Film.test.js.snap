// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film render Film correctly 1`] = `
<div
  className="col-md-4 mt-5"
>
  <Link
    to="/movie/1"
  >
    <img
      alt="text"
      src="https://m.media-amazon.com/images/M/MV5BNGNhMDIzZTUtNTBlZi00MTRlLWFjM2ItYzViMjE3YzI5MjljXkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SY1000_CR0,0,686,1000_AL_.jpg"
    />
  </Link>
  <div
    className="row mt-3"
  >
    <div
      className="col-6"
    >
      <span>
        film 1
      </span>
    </div>
    <div
      className="col-2 ml-auto mr-3"
    >
      <span
        className="age"
      >
        1332
      </span>
    </div>
  </div>
  <div>
    <span>
      genre 1
    </span>
  </div>
</div>
`;
